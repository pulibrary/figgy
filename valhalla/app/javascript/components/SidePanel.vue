<template>
  <div class="sidePanel actions">
    <!-- Resource Form-->
    <resource-form v-if="selectedTotal === 0" id="noneSelected" class="formContent"/>
    <!-- Multiple Selected Form-->
    <filesets-form v-if="selectedTotal > 1" id="multiSelected"/>
    <!-- Single Selected Form-->
    <fileset-form v-if="selectedTotal === 1" id="singleSelected"/>
    <controls></controls>
  </div>
</template>

<script>
import ResourceForm from './ResourceForm'
import FilesetForm from './FilesetForm'
import FilesetsForm from './FilesetsForm'
import Controls from './Controls'

export default {
  name: 'panel',
  components: {
    'resource-form': ResourceForm,
    'fileset-form': FilesetForm,
    'filesets-form': FilesetsForm,
    Controls
  },
  computed: {
    selectedTotal () {
      return this.$store.state.selected.length
    }
  }
}
</script>
