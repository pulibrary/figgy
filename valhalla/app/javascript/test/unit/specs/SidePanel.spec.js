import 'babel-polyfill'
import Vuex from 'vuex'
import { shallow, createLocalVue } from 'vue-test-utils'
import SidePanel from '@/components/SidePanel'
const localVue = createLocalVue()
localVue.use(Vuex)

describe('SidePanel.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(SidePanel, {
      computed: {
        selectedTotal () {
          return 2
        }
      },
      localVue
    })
  })

  it('selectedTotal function has the expected value', () => {
    expect(wrapper.vm.selectedTotal).toBe(2)
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

})
