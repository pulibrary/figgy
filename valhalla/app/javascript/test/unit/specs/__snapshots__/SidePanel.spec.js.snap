// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel.vue has the expected html structure 1`] = `
<div
  class="sidePanel actions"
>
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
</div>
`;
